Volontulo/app/src/main/java/com/stxnext/volontulo/logic/im/IMService.java

```diff
@@ -0,0 +1,208 @@
+package com.stxnext.volontulo.logic.im;
+
+import android.app.Service;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.Binder;
+import android.os.IBinder;
+import android.support.annotation.Nullable;
+import android.support.v4.content.LocalBroadcastManager;
+import android.text.TextUtils;
+import android.util.Log;
+
+import com.sinch.android.rtc.ClientRegistration;
+import com.sinch.android.rtc.PushPair;
+import com.sinch.android.rtc.Sinch;
+import com.sinch.android.rtc.SinchClient;
+import com.sinch.android.rtc.SinchClientListener;
+import com.sinch.android.rtc.SinchError;
+import com.sinch.android.rtc.messaging.Message;
+import com.sinch.android.rtc.messaging.MessageClient;
+import com.sinch.android.rtc.messaging.MessageClientListener;
+import com.sinch.android.rtc.messaging.MessageDeliveryInfo;
+import com.sinch.android.rtc.messaging.MessageFailureInfo;
+import com.sinch.android.rtc.messaging.WritableMessage;
+import com.stxnext.volontulo.BuildConfig;
+import com.stxnext.volontulo.logic.im.config.ImConfigFactory;
+import com.stxnext.volontulo.logic.im.config.ImConfiguration;
+
+import java.util.List;
+
+import io.realm.Realm;
+
+public class IMService extends Service implements SinchClientListener {
```

Comment:
Compare names: IMService, ImConfiguration. It should be one convetion: IMService & IMConfiguration or ImService and ImConfiguration.